core/model/upgradeseries.go

```diff
@@ -7,6 +7,83 @@ import (
 	"github.com/juju/errors"
 )
 
+// UpgradeSeriesStatus is a status that a machine or unit can be in during the
+// execution of an OS series upgrade on its host machine.
+type UpgradeSeriesStatus int
+
+// String returns an indicative description of the status.
+func (s UpgradeSeriesStatus) String() string {
+	return upgradeSeriesStatusNames[s]
+}
+
+// Enumerating the status values allows us to represent relative progression
+// though the upgrade-series workflow.
+const (
+	UpgradeSeriesNotStarted UpgradeSeriesStatus = iota
+	UpgradeSeriesPrepareStarted
+	UpgradeSeriesPrepareMachine
+	UpgradeSeriesPrepareComplete
+	UpgradeSeriesCompleteStarted
+	UpgradeSeriesComplete
+	UpgradeSeriesError
+)
+
+var upgradeSeriesStatusNames = map[UpgradeSeriesStatus]string{
+	UpgradeSeriesNotStarted:      "not started",
+	UpgradeSeriesPrepareStarted:  "prepare started",
+	UpgradeSeriesPrepareMachine:  "prepare machine",
+	UpgradeSeriesPrepareComplete: "prepare complete",
+	UpgradeSeriesCompleteStarted: "complete started",
+	UpgradeSeriesComplete:        "complete",
+	UpgradeSeriesError:           "error",
+}
+
+// TODO (manadart 2018-08-07) Everything below here to be discarded.
+
+// ToOldStatus is a shim between the new and old status representations.
+// It exists to bootstrap that transition. Unit tests omitted by design.
+func (s UpgradeSeriesStatus) ToOldStatus() (UpgradeSeriesStatusType, MachineSeriesUpgradeStatus) {
```

Comment:
Currently the machine statuses are only checked  in few places. Do you think It may be easier just to replace them entirely rather than construct this adapter? However, I do understand that his adapter can just simply and easily be removed once the new states are replaced.

Reply:
It's less that they are used in few places.

More that I can add all the usages I need without thinking about implications of changing the lock data structure, and none of those places outside of state need to change afterwards.